flink-core/src/main/java/org/apache/flink/util/NetUtils.java

```diff
@@ -116,6 +119,39 @@ private static URL validateHostPortString(String hostPort) {
         }
     }
 
+    /**
+     * Calls {@link ServerSocket#accept()} on the provided server socket, suppressing any thrown
+     * {@link SocketTimeoutException}s. This is a workaround for the underlying JDK-8237858 bug in
+     * JDK 11 that can cause errant SocketTimeoutExceptions to be thrown at unexpected times.
+     *
+     * <p>This method expects the provided ServerSocket has no timeout set (SO_TIMEOUT of 0),
+     * indicating an infinite timeout. It will suppress all SocketTimeoutExceptions, even if a
+     * ServerSocket with a non-zero timeout is passed in.
+     *
+     * @param serverSocket a ServerSocket with {@link SocketOptions#SO_TIMEOUT SO_TIMEOUT} set to 0;
+     *     if SO_TIMEOUT is greater than 0, then this method will suppress SocketTimeoutException;
+     *     must not be null
+     * @return the new Socket
+     * @exception IOException if an I/O error occurs when waiting for a connection.
+     * @exception SecurityException if a security manager exists and its {@code checkAccept} method
```

Comment:
We shouldn't copy code from the JDK verbatim. Let's just add a reference to `ServerSocket#accept`.